tests/test_todo.py

```diff
@@ -0,0 +1,60 @@
+from unittest import TestCase
+
+from caldav.objects import Todo
+
+
+# example from http://www.rfc-editor.org/rfc/rfc5545.txt
+VTODO = """BEGIN:VCALENDAR
+VERSION:2.0
+PRODID:-//Example Corp.//CalDAV Client//EN
+BEGIN:VTODO
+UID:20070313T123432Z-456553@example.com
+DTSTAMP:20070313T123432Z
+DUE;VALUE=DATE:20070501
+SUMMARY:Submit Quebec Income Tax Return for 2006
+CLASS:CONFIDENTIAL
+CATEGORIES:FAMILY,FINANCE
+STATUS:NEEDS-ACTION
+END:VTODO
+END:VCALENDAR"""
+
+
+## a todo without uid.  Should it be possible to store it at all?
+VTODO_NO_UID = """
+BEGIN:VCALENDAR
+VERSION:2.0
+PRODID:-//Example Corp.//CalDAV Client//EN
+BEGIN:VTODO
+DTSTAMP:19980101T130000Z
+DTSTART:19980415T133000Z
+DUE:19980516T045959Z
+SUMMARY:Get stuck with Netfix and forget about the tax income declaration
+CLASS:CONFIDENTIAL
+CATEGORIES:FAMILY
+PRIORITY:1
+END:VTODO
+END:VCALENDAR"""
+
+
+class TodoTestCase(TestCase):
+
+    def test_it_sets_the_id(self):
+        todo = Todo(data=VTODO)
+
+        self.assertEqual(todo.id, "20070313T123432Z-456553@example.com")
+
+    def test_it_prefers_the_id_of_the_vtodo(self):
+        # Is this actually the expected behaviour?
```

Comment:
This seems reasonable to me, but there may be some reason to prefer the `id` that's passed in to the constructor.

Reply:
I think that if the caller sends one id as parameter to __init__() and another id implicitly in the oject passed, then the caller is doing something wrong, and perhaps an exception would be more appropriate?